src/pkg/cli/client/byoc/aws/byoc.go

```diff
@@ -379,6 +379,7 @@ func (b *ByocAws) PrepareDomainDelegation(ctx context.Context, req client.Prepar
 		if zone != nil {
 			zoneId = zone.Id
 		}
+		// TODO: avoid creating the delegation set if we're in preview mode
```

Comment:
Do we want to create a mock delegation set in preview mode?

Reply:
We could, but I don't think it's worth it. Delegation Sets are free and don't even show up in the AWS console.